specreduce/extract.py

```diff
@@ -3,216 +3,162 @@
-        j1 = self._find_nearest_int(low_end)
-        j2 = self._find_nearest_int(high_end)
-
-        # fractional pixel areas at the end points
-        s1 = 0.5 - (low_end - j1)
-        s2 = 0.5 + high_end - j2
-
-        # add up the total flux around the trace_line
-        extracted_result[wave_index] = np.nansum(image[j1 + 1:j2, wave_index])
-        extracted_result[wave_index] += np.nansum(image[j1, wave_index]) * s1
-        extracted_result[wave_index] += np.nansum(image[j2, wave_index]) * s2
-
-    def _find_nearest_int(self, end_point):
-        if (end_point % 1) < 0.5:
-            return int(end_point)
-        else:
-            return int(end_point + 1)
-
-    def get_checkplot(self):
-        trace_line = self.last_trace.line
-
-        fig = plt.figure()
-        plt.imshow(self.last_img, origin='lower', aspect='auto', cmap=plt.cm.Greys_r)
-        plt.clim(np.percentile(self.last_img, (5, 98)))
-
-        plt.plot(np.arange(len(trace_line)), trace_line, c='C0')
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line + self.apwidth,
-            trace_line - self.apwidth,
-            color='C0',
-            alpha=0.5
-        )
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line + self.apwidth + self.skysep,
-            trace_line + self.apwidth + self.skysep + self.skywidth,
-            color='C1',
-            alpha=0.5
-        )
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line - self.apwidth - self.skysep,
-            trace_line - self.apwidth - self.skysep - self.skywidth,
-            color='C1',
-            alpha=0.5
-        )
-        plt.ylim(
-            np.min(
-                trace_line - (self.apwidth + self.skysep + self.skywidth) * 2
-            ),
-            np.max(
-                trace_line + (self.apwidth + self.skysep + self.skywidth) * 2
-            )
-        )
-
-        return fig
+    # TODO: what is a reasonable default?
+    # TODO: int or float?
+    width: int = 5
+
+    # TODO: should disp_axis and crossdisp_axis be defined in the Trace object?
+
+    def __call__(self, image, trace_object, disp_axis=1, crossdisp_axis=(0,)):
+        """
+        Extract the 1D spectrum using the boxcar method.
+
+        Parameters
+        ----------
+        image : nddata-compatible image
+            image with 2-D spectral image data
+        trace_object : Trace
+            object with the trace
+        disp_axis : int
+            dispersion axis
+        crossdisp_axis : tuple (to support both 2D and 3D data)
+            cross-dispersion axis
+
+
+        Returns
+        -------
+        spec : `~specutils.Spectrum1D`
+            The extracted 1d spectrum expressed in DN and pixel units
+        """
+        # this check only applies to FlatTrace instances
+        if hasattr(trace_object, 'trace_pos'):
+            self.center = trace_object.trace_pos
```

Comment:
`self.center` isn't currently used anywhere except the following check... do we need to set it?  And should we update the `ValueError` to mention `trace_object.trace_pos` instead since that is the value that the user could actually change?

Reply:
No, we don't need to set `self.center`. I did it in this way just so the existing code could be reused.